common/common-protobuf/src/main/java/org/apache/servicecomb/codec/protobuf/definition/ProtobufManager.java

```diff
@@ -45,13 +50,28 @@
   private ProtobufManager() {
   }
 
+  /**
+   * only for app classloader
+   * @return
+   */
+  public static ScopedProtobufSchemaManager getDefaultScopedProtobufSchemaManager() {
+    return defaultScopedProtobufSchemaManager;
+  }
+
   public static OperationProtobuf getOrCreateOperation(OperationMeta operationMeta) throws Exception {
-    OperationProtobuf operationProtobuf = operationMeta.getExtData(ProtobufManager.EXT_ID);
+    OperationProtobuf operationProtobuf = operationMeta.getExtData(EXT_ID);
```

Comment:
if the ExtData is not volatile,  and there are two thread a modify it, you may face some issue here.

Reply:
extData is a ConcurrentHashMap, no problem.